internal/app/triggermessageprocessor.go

```diff
@@ -0,0 +1,143 @@
+	return container.SecretProviderFrom(b.Service.dic.Get)
+}
+
+func NewTriggerServiceBinding(svc *Service) trigger.ServiceBinding {
+	return &simpleTriggerServiceBinding{
+		svc,
+		svc.runtime,
+	}
+}
+
+func (b *simpleTriggerServiceBinding) DIC() *di.Container {
+	return b.Service.dic
+}
+
+func (b *simpleTriggerServiceBinding) BuildContext(env types.MessageEnvelope) interfaces.AppFunctionContext {
+	return appfunction.NewContext(env.CorrelationID, b.Service.dic, env.ContentType)
+}
+
+func (b *simpleTriggerServiceBinding) Config() *common.ConfigurationStruct {
+	return b.config
+}
+
+// customTriggerBinding wraps the CustomTriggerServiceBinding interface so that we can attach methods
+type triggerMessageProcessor struct {
+	bnd trigger.ServiceBinding
+}
+
+// Process provides runtime orchestration to pass the envelope / context to the pipeline.
+// Deprecated: This does NOT support multi-pipeline usage.  Will send a message to the default pipeline ONLY and throw if not configured.  Use MessageReceived.
+func (mp *triggerMessageProcessor) Process(ctx interfaces.AppFunctionContext, envelope types.MessageEnvelope) error {
+	context, ok := ctx.(*appfunction.Context)
+	if !ok {
+		return fmt.Errorf("App Context must be an instance of internal appfunction.Context. Use NewAppContext to create instance.")
+	}
+
+	defaultPipelines := mp.bnd.GetMatchingPipelines(interfaces.DefaultPipelineId)
+
+	if len(defaultPipelines) != 1 {
+		return fmt.Errorf("TriggerMessageProcessor does not support multiple pipelines.  Please use TriggerMessageHandler for multi-pipeline support")
+	}
+
+	messageError := mp.bnd.ProcessMessage(context, envelope, defaultPipelines[0])
+	if messageError != nil {
+		// ProcessMessage logs the error, so no need to log it here.
+		return messageError.Err
+	}
+
+	return nil
+}
+
+// MessageReceived provides runtime orchestration to pass the envelope / context to configured pipeline(s).
+func (mp *triggerMessageProcessor) MessageReceived(ctx interfaces.AppFunctionContext, envelope types.MessageEnvelope, outputHandler interfaces.PipelineResponseHandler) error {
+	return mp.messageReceived(ctx, envelope, outputHandler)
+}
+
+func (mp *triggerMessageProcessor) messageReceived(ctx interfaces.AppFunctionContext, envelope types.MessageEnvelope, responseHandler interfaces.PipelineResponseHandler) error {
+	lc := mp.bnd.LoggingClient()
+
+	lc.Tracef("custom trigger attempting to find pipeline(s) for topic %s", envelope.ReceivedTopic)
+
+	// ensure we have a context established that we can safely cast to *appfunction.Context to pass to runtime
+	if _, ok := ctx.(*appfunction.Context); ctx == nil || !ok {
+		ctx = mp.bnd.BuildContext(envelope)
+	}
+
+	pipelines := mp.bnd.GetMatchingPipelines(envelope.ReceivedTopic)
+
+	lc.Tracef("trigger found %d pipeline(s) that match the incoming topic '%s'", len(pipelines), envelope.ReceivedTopic)
+
+	if responseHandler == nil && len(pipelines) != 0 {
```

Comment:
should this be?
```suggestion
	if responseHandler == nil && len(pipelines) != 1 {
```

Reply:
need to think about this I think the message might just be wrong.

Reply:
I think this is just a remove.  Passing a nil response handler should probably be allowed, even if its a little weird.